modules/promotion/src/Plugin/Commerce/PromotionOffer/PercentageOffBase.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Drupal\commerce_promotion\Plugin\Commerce\PromotionOffer;
+
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Provides base class for percentage off offers.
+ */
+abstract class PercentageOffBase extends PromotionOfferBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [
+      'amount' => 0,
+    ] + parent::defaultConfiguration();
+  }
+
+  /**
+   * Gets the percentage amount, as a decimal, negated.
+   *
+   * @return string
+   *   The amount.
+   */
+  public function getAmount() {
+    return (string) (-$this->configuration['amount']);
```

Comment:
Should the negation be done here, or in the code that creates the adjustment? Isn't it saner to say "This class provides a 19% off promotion" instead of "This class provides a -19% off promotion"?